ament_clang_tidy/ament_clang_tidy/main.py

```diff
@@ -377,5 +388,105 @@ def get_xunit_content(report, testname, elapsed):
     return xml
 
 
+def get_clang_tidy_version(clang_tidy_bin):
+    version = ''
+    try:
+        cmd_args = [clang_tidy_bin, '--version']
+        output = subprocess.check_output(cmd_args).decode()
+        m = re.search('.*LLVM version (.*)\n', output)
+        if m:
+            version = m.group(1)
+    except Exception as e:
+        print(f'Failed to get the clang tidy version number: {e}', file=sys.stderr)
```

Comment:
```suggestion
        print(f"Failed to get the version from '{' '.join(cmd_args)}': {e}", file=sys.stderr)
```

Should this throw? Is it ok to just return `''` when the version cannot be determined? Also, you end up returning `''` if the regex doesn't match, so same question in that case.

Reply:
In the unlikely case that we can't get the clang tidy version number, I think it's OK to proceed. I've updated the code to return 'unknown' instead of the empty string and warn if either the regex fails or the clang_tidy invocation fails.